lib/puppet-strings/markdown/base.rb

```diff
@@ -50,6 +50,33 @@ module PuppetStrings::Markdown
   #  ") inherits foo::bar {\n" +
   #  "}"}
   class Base
+    # Set or return the name of the group
+    #
+    # @param [Optional[String]] Name of the group to set
+    # @return [String] Name of the group
+    def self.group_name(name = nil)
```

Comment:
Thoughts about splitting the functionality in `self.group_name()` and `self.group_name=(name)`? Would that work?

Same comment applies to `yard_types`.

Reply:
I actually tried doing

```ruby
class << self
  attr_accessor :group_name
  attr_accessor :yard_types
end
```

but I ran into a few problems. One was that it interpreted `yard_types = [...]` as trying to set a local variable instead of calling a class method, but maybe I could have called `self.yard_types = [...]`.

I wanted it to look more like functions like `attr_accessor` since I figured that would feel more natural, but it did seem inconsistent with the way Ruby does things elsewhere.

I don’t have a real good sense for the Ruby Way anymore… been writing other languages for too long.

Reply:
Fair enough. I suppose this is good enough.